nixos/modules/programs/steam.nix

```diff
@@ -4,17 +4,31 @@ with lib;
 
 let
   cfg = config.programs.steam;
-
-  steam = pkgs.steam.override {
-    extraLibraries = pkgs: with config.hardware.opengl;
-      if pkgs.hostPlatform.is64bit
-      then [ package ] ++ extraPackages
-      else [ package32 ] ++ extraPackages32;
-  };
 in {
   options.programs.steam = {
     enable = mkEnableOption (lib.mdDoc "steam");
 
+    package = mkOption {
+      type        = types.package;
+      default     = pkgs.steam.override {
+        extraLibraries = pkgs: with config.hardware.opengl;
+          if pkgs.hostPlatform.is64bit
+          then [ package ] ++ extraPackages
+          else [ package32 ] ++ extraPackages32;
+      };
+      defaultText = literalExpression ''
+        pkgs.steam.override {
```

Comment:
does it make sense for the user to disable this overrride? otherwise we could apply it by default afterwards. (so the user only has to specify steam.package = customsteam; and the module will handle the opengl override)

Reply:
I can imagine that you'd like to override which kind of opengl library steam should use compared to the rest of the system. I don't have any practical use-case atm, but something like having a beta version of a vulkan driver for steam vs a stable one for the rest of the system. I'm just making this up, but it does seem viable.

Reply:
ah, ok, then i'll merge this as-is